protocols/v2/roles-logic-sv2/src/job_creator.rs

```diff
@@ -285,11 +285,124 @@ fn coinbase(
     }
 }
 
+/// Helper type to strip a segwit data from the coinbase_tx_prefix and coinbase_tx_suffix
+/// to ensure miners are hashing with the correct coinbase
+pub fn extended_job_to_non_segwit(
+    job: NewExtendedMiningJob<'static>,
```

Comment:
this is an only Sv1 feature, Sv2 software should not depend on upstream removing segwit data from the tx. Also it return a `NewExtendedMiningJob` that do not complain with the spec. If possible I would put this function somewhere else maybe in `utils/something` or in `protocols/sv1`, and I would also modify the signature so that it do not return a `NewExtendedMiningJob` but something else maybe just `(prefix,suffix)` ?

Reply:
Okay, I can do this, but first I would like to confirm that  all SV2 miner firmware is segwit compatible. If not we will have issues later on, and if that is the case, this function could be used in the pool before broadcasting to ensure support for either cases. Assuming you mean its not compatible with the spec since the job is getting altered after the pool sends it out?

Reply:
ok maybe make more sense to apply your patch  everywhere both on Sv1 and Sv2. In the extended job we send the tx_outputs so no sense to send also the witness.